evalml/pipelines/components/transformers/imputers/simple_imputer.py

```diff
@@ -69,42 +64,33 @@ def fit(self, X, y=None):
         if set([lt.type_string for lt in X.ww.logical_types.values()]) == {
             "boolean",
             "categorical",
-        } and not all(
-            [
-                is_categorical_actually_boolean(X, col)
-                for col in X.ww.select("Categorical")
-            ],
-        ):
+        }:
             raise ValueError(
                 "SimpleImputer cannot handle dataframes with both boolean and categorical features.  Use Imputer instead.",
             )
 
         nan_ratio = X.isna().sum() / X.shape[0]
         self._all_null_cols = nan_ratio[nan_ratio == 1].index.tolist()
 
-        X, _ = drop_natural_language_columns(X)
+        # Keep track of the different types of data in X
+        self._all_null_cols = nan_ratio[nan_ratio == 1].index.tolist()
+        self._natural_language_cols = X.ww.select("NaturalLanguage").columns.to_list()
 
-        # Convert any boolean columns to IntegerNullable, but keep track of the columns so they can be converted back
-        self._boolean_cols = list(
-            X.ww.select(
-                include=["Boolean", "BooleanNullable"],
-                return_schema=True,
-            ).columns,
-        )
-        # Make sure we're tracking Categorical columns that should be boolean as well
-        self._boolean_cols.extend(
-            [
-                col
-                for col in X.ww.select("Categorical")
-                if is_categorical_actually_boolean(X, col)
```

Comment:
Explanation as to why we can remove `is_categorical_actually_boolean`:
- Was originally added in https://github.com/alteryx/evalml/pull/3525 when we first started imputing boolean columns separately from categorical columns.
- At that time, the current nullable type handling involved converting BooleanNullable columns to Categorical, so this logic was needed to correctly partition columns for imputing. We no longer create these categorical booleans (I caught and removed the last case where we were doing it in the EmailFeaturizer), and I don't think we need to handle the case where a user has created them, so uses of this should be removed.